File: front/android/app/src/main/java/com/app/TmapViewManager.java
package com.app;

import android.content.Context;
import com.facebook.react.uimanager.SimpleViewManager;
import com.facebook.react.uimanager.ThemedReactContext;
import com.skt.tmap.TMapView;  // 올바른 경로로 import

public class TmapViewManager extends SimpleViewManager<TMapView> {

    @Override
    public String getName() {
        return "TmapView";
    }

    @Override
    protected TMapView createViewInstance(ThemedReactContext reactContext) {
        TMapView tMapView = new TMapView((Context) reactContext);
        tMapView.setSKTMapApiKey("N58gCr0OpV7gn4udSAHyC3PZyY2HC7Jt8e4LQ5WB");
        return tMapView;
    }
}

File: front/android/app/src/main/java/com/app/TMapViewManager.kt
package com.app

import android.util.Log
import com.app.views.TMapNativeView
import com.facebook.react.bridge.Arguments
import com.facebook.react.bridge.WritableMap
import com.facebook.react.common.MapBuilder
import com.facebook.react.uimanager.SimpleViewManager
import com.facebook.react.uimanager.ThemedReactContext
import com.facebook.react.uimanager.events.RCTEventEmitter


class TMapViewManager : SimpleViewManager<TMapNativeView>() {

    override fun getName(): String {
        return "TMapView"
    }

    override fun createViewInstance(reactContext: ThemedReactContext): TMapNativeView {
        val view = TMapNativeView(reactContext)

        view.post {
            view.setMapReadyListener {
                Log.d("TMapManager", "Listener 등록 성공 from Manager")
            }
        }

        return view
    }
}
